Fischer_algorithm_Anca/test.c

```diff
@@ -1,56 +1,89 @@
 #include <stdio.h>
 
-#include "hashset.h"
-#include "hashset_itr.h"
+
 #include "structures.h"
 #include "fonctions_fischer.h"
 
+
 int main()
 {
-    //test creation set points 
-    printf("Test creation set points: \n");
-
-    POINT p1 = {10,15};
-    POINT p2 = {10,5};
-    POINT p3 = {5,10};
-    POINT p4 = {15,10};
-    POINT p5 = {0,0};
-    POINT *items[] = {&p1,&p2,&p3,&p4,NULL}; 
-    size_t ii, nitems = 4;
-    hashset_t set = hashset_create();
-    hashset_itr_t iter = hashset_iterator(set);
-
-    if (set == NULL) {
-        fprintf(stderr, "failed to create hashset instance\n");
-        abort();
-    }
-
-    for (ii = 0; ii < nitems; ++ii) {
-        hashset_add(set, items[ii]);
-    }
-
-    for (ii = 0; ii < nitems; ++ii) {
-        printf("%d\n",hashset_is_member(set, items[ii]));
-    }
-
-    printf("%d\n",hashset_is_member(set,&p5));
-    
-    printf("Le set contient %d points\n", (int)set->nitems);
+
+    CERCLE vraiC; //Solution universelle pour les 3 tests
+    vraiC.x = 10.0; vraiC.y=10.0; vraiC.d=10.0; 
+    int nbPoints;
+
+    // Test 1 : Points cocirculaires
+
     
-    //test iterateur set points
-    printf("Test iterateur set points: \n");
-
-    while(hashset_iterator_has_next(iter))
-    {
-        
-        printf("%d %d\n",((POINT*)hashset_iterator_value(iter))->x, ((POINT*)hashset_iterator_value(iter))->y);
-        hashset_iterator_next(iter);
-    }
+    printf("\n** Test 1 : Points cocirculaires **\n\n");
+
+    POINT tab1[] = {{10.0,15.0},{10.0,5.0},{5.0,10.0},{15.0,10.0}}; 
+    nbPoints = sizeof(tab1) / sizeof (tab1[0]);
+
+    printf("Tableau donnee: ");
+    for(int i=0;i<nbPoints;i++)
```

Comment:
La variable i doit être déclaré à l'extérieur en C90: 

test.c:24:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for(int i=0;i<nbPoints;i++)
     ^

(c'est aussi en de nombreux autres endroits)